src/components/Registration/Manual/SearchAttendee.vue

```diff
@@ -1,155 +1,268 @@
+import { FunnelIcon } from '@heroicons/vue/24/outline'
 
-const query = ref('')
-const typing = ref(false)
+// INITIALISE TEMPLATE REFS
 const searchBar = ref(null)
+const refs = ref([])
 
-const erase = () => {
-  query.value = ''
-}
-
-watch(query, () => {
-  if (query.value !== '') typing.value = true
-  else typing.value = false
+onMounted(() => {
+  refs.value = Array(filterOptions.length)
+    .fill(null)
+    .map(() => ref(null))
 })
 
-const checkIn = (item) => {
-  item.checkedIn.value = true
+function setRef(index) {
+  return (el) => {
+    refs.value[index] = el
+  }
 }
 
+const menuOpen = ref(false)
+const query = ref('')
+
 const items = [
   {
     id: 1,
     name: 'Wei Tat Chung',
-    position: 'Chairperson',
-    role: 'Organiser',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 2,
-    name: 'Shaun Laclemence',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Don Chia',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Shaun Ming Laclemence',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    id: 4,
+    name: 'Very very very very very very very very very very long name',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
+]
+
+const filterOptions = [
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
-    email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    id: 'filterRole',
+    name: 'Role',
+    show: ref(false)
+  },
+  {
+    id: 'filterMem',
+    name: 'Member type',
+    show: ref(false)
+  },
+  {
+    id: 'filterOrg',
+    name: 'Organisation',
+    show: ref(false)
   }
 ]
+
+// toggles menuOpen and updates filter options when menu is open
+const updateMenu = () => {
+  new Promise((resolve) => {
+    menuOpen.value = !menuOpen.value
+    resolve()
+  }).then(() => {
+    if (menuOpen.value) {
+      refs.value.forEach((element, index) => element.checked = filterOptions[index].show.value)
```

Comment:
why not just use :checked prop at the input field to call for the show boolean since you already have it in v-for?

Reply:
All previous comments resolved.